pkg/binding/spec/spec.go

```diff
@@ -108,7 +108,7 @@ func (v *version) SetAttribute(c ce.EventContextWriter, name string, value inter
 	name = strings.ToLower(name)
 	var err error
 	if strings.HasPrefix(name, v.prefix) { // Extension attribute
-		value, err = types.Validate(value)
+		value, err = types.ValidateAndWrap(value)
```

Comment:
I would be inclined to just change the Validate() function rather than introduce a new function. The purpose of Validate() is to convert any "convertible" value into the "standard" Go form. As you've pointed out the "standard" form form URL and Time values should be the type wrappers since the native types lack proper marshalling - so you're fixing a bug in Validate(), rather than adding a new feature. I'd update the doc comments in the types package to reflect that.

Reply:
My bad, I thought `types.Validate` was used somewhere else